Kconfig.zephyr

```diff
@@ -324,6 +324,14 @@ config LINKER_USE_RELAX
 	  Disabling it can reduce performance, as the linker is no longer able to substiture long /
 	  in-effective jump calls to shorter / more effective instructions.
 
+config LINKER_ALLOW_RWX_SEGMENTS
+	bool "Allow RWX output segments"
+	default n
```

Comment:
i would default this to y, given this issue is with any target we build using gcc 14.2

Reply:
at least it's inconsistent with the commit message, so can result in confusion if someone later does a `git blame` and want to understand why this was made.
> This is set to 'y' by default as nearly all Zephyr targets do not yet mark
> all data destined to land in ROM as constant.

Reply:
I'm thinking it would make more sense to flip the sense of the condition so that a 'true' value enables these warnings rather than suppresses them -- `CONFIG_LINKER_WARN_RWX_SEGMENTS`. Seem reasonable?